msi/msi_test.go

```diff
@@ -70,7 +70,91 @@ func TestCanGetMsi(t *testing.T) {
 	if err != nil {
 		t.Fatal(err.Error())
 	}
-	err = ioutil.WriteFile(fmt.Sprintf("%s/msi.json", outdir), msiJsonBytes[:], 0700)
+	err = ioutil.WriteFile(fmt.Sprintf("%s/msi1.json", outdir), msiJsonBytes[:], 0700)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+}
+
+func TestCanGetMsiForStorage(t *testing.T) {
+	t.Skip() // for testing on Azure VM only
+	outdir := "./testoutput"
+	os.Mkdir(outdir, 0777)
+	secureHttpClient := httputil.NewSecureHttpClient(httputil.NoRetry)
+	msiProvider := NewMsiProvider(secureHttpClient)
+	msi, err := msiProvider.GetMsiForResoruce("https://storage.azure.com/")
```

Comment:
No assertion on the msi you get in any of these tests? If it on purpose?

Reply:
These are not true unit tests. They are marked as t.skip() because we need to provision VM's in azure with system/user assigned identities to be able to properly test the msi.

I unskip these test and eyeball the output msi.json to make sure the changes are working.